ydb/core/tx/schemeshard/schemeshard_import.cpp

```diff
@@ -106,36 +105,26 @@ namespace {
         }
         Y_ABORT_UNLESS(indexIdx < item.Table->indexes_size());
 
-        const ui32 partsTotal = ss->Tables.contains(item.DstPathId) ?
-            ss->Tables.at(item.DstPathId)->GetPartitions().size() :
-            GetTablePartsFromRequest(*item.Table);
+        const ui32 partsTotal = GetTablePartsFromRequest(*item.Table);
 
         const auto buildUid = MakeIndexBuildUid(importInfo, itemIdx, indexIdx);
         if (ss->IndexBuildsByUid.contains(buildUid)) {
-            const auto& indexBuild = ss->IndexBuildsByUid.FindPtr(buildUid)->Get();
-
-
-            ui32 partsCompleted;
-            if (indexBuild->State < TIndexBuildInfo::EState::Filling) {
-                // Shards are not filled yet
-                partsCompleted = 0;
-            } else if (indexBuild->State < TIndexBuildInfo::EState::Applying) {
-                // Shards are not processed yet
-                partsCompleted = static_cast<ui32>(indexBuild->CalcProgressPercent() / 100.0f * partsTotal );
-            } else {
-                // All shards are processed
+            const auto& indexBuild = ss->IndexBuildsByUid[buildUid];
+
+            ui32 partsCompleted = 0;
+            if (indexBuild->IsTransferring()) {
+                partsCompleted = static_cast<ui32>(indexBuild->CalcProgressPercent() / 100.0f * partsTotal);
+            } else if (indexBuild->IsApplying() || indexBuild->IsFinished()) {
                 partsCompleted = partsTotal;
             }
 
             itemProgress.set_parts_total(itemProgress.parts_total() + partsTotal);
             itemProgress.set_parts_completed(itemProgress.parts_completed() + partsCompleted);
-            *itemProgress.mutable_end_time() = SecondsToProtoTimeStamp(indexBuild->EndTime.Seconds());
-        } else {
-            auto partsCompleted = indexIdx >= item.NextIndexIdx ? 0 : partsTotal;
-
+            if (indexIdx == item.Table->indexes_size() - 1 && indexBuild->IsFinished()) {
```

Comment:
Тут предлагаю просто проверить, что EndTime не равен нулю, и также его безусловно выставить. Более новые индексы его перезапишут или обнулят.
```suggestion
            if (indexBuild->IsFinished()) {
```